src/Illuminate/Database/Eloquent/FactoryBuilder.php

```diff
@@ -131,7 +131,22 @@ protected function makeInstance(array $attributes = [])
 
             $definition = call_user_func($this->definitions[$this->class][$this->name], $this->faker, $attributes);
 
-            return new $this->class(array_merge($definition, $attributes));
+            $evaluated = $this->evaluateClosures(array_merge($definition, $attributes));
+
+            return new $this->class($evaluated);
         });
     }
+
+    /**
+     * Delay closures evaluation after merging to avoid duplication.
+     *
+     * @param  array  $attributes
+     * @return array
+     */
+    protected function evaluateClosures(array $attributes)
```

Comment:
I agree with @devinfd otherwise one might wonder why we're evaluating closures in the first place.